lib/money/money.rb

```diff
@@ -269,29 +271,33 @@ def allocate(splits)
   #   Money.new(100).allocate_max_amounts([Money.new(5), Money.new(2)])
   #     #=> [Money.new(5), Money.new(2)]
   def allocate_max_amounts(maximums)
-    maximums = maximums.map { |max_amount| max_amount.to_money }
-    maximums_total = maximums.sum
+    allocate_currency = extract_currency(maximums + [self])
+    maximums = maximums.map { |max| max.to_money(allocate_currency) }
```

Comment:
If the amount is already a money object it won't do anything right?

Reply:
correct 👍 , this will only convert integer and strings to money objects.